packages/devops/scripts/deployment/startBackEnds.sh

```diff
@@ -24,6 +25,10 @@ for PACKAGE in ${PACKAGES[@]}; do
             # reset cwd back to `/tupaia`
             cd ${TUPAIA_DIR}
 
+            # ensure db master password is set correctly
+            echo "Setting RDS master password for ${DEPLOYMENT_NAME}"
+            source packages/data-api/.env && aws rds modify-db-instance --db-instance-identifier $DEPLOYMENT_NAME --master-user-password $DB_PG_PASSWORD
```

Comment:
Not quite happy with this, but keen to get some input from the reviewer.

Basically, when an RDS instance is spun up, it will have a random generated master user password. So we need to manually set it to be our postgres user password. The postgres user password is stored in the `.env` files, so we need to be in a context that has access to those. Unfortunately that means we can't do this within the lambda scripts, and instead need to do it in the startup process.

Overall I've come to accept that, and this place seems appropriate to me, however I'm still not quite happy with it. We're `source`ing the `data-api` environment variables in order to get the postgres user password which feels really ugly...

My current plan is to make this a node script in the database package, something like `set-rds-master-password`. So this would be: `yarn workspace @tupaia/database set-rds-master-password $DEPLOYMENT_NAME`

I still don't love it though, as it pollutes our generally infrastructure independent `database` package code with logic solely for RDS... Was hoping maybe the reviewer had a cleverer solution?

Reply:
There's one other option I can think of, which also has its downsides: add the db password as a secret on parameter store, and have the lambda scripts grab it from there during db setup. The downsides are it exposes the secret to another place (lambda), and means there's an unusual spot to update if we ever need to. Not terrible, but I don't think it's any better or worse than what you've already got, so might as well stick with that.

That does get me thinking about the automatic password rotation provided by AWS Secrets Manager, which integrates directly with RDS...there could be a nice way of using that, and then the tricky bit becomes setting up a process that gets the rotated password through into our environment variables. A future task though :-)

Reply:
I like your suggestion, but I'm with ya that it's kinda on par with the suggestion I had, and I feel I'll have an easier time implementing mine.

Definitely with you on the rotating password though... which will be tricky be rather valuable to get working!